pandas/tests/interchange/test_impl.py

```diff
@@ -394,6 +392,13 @@ def test_large_string():
     tm.assert_frame_equal(result, expected)
 
 
+def test_non_str_names():
+    # https://github.com/pandas-dev/pandas/issues/56701
+    df = pd.Series([1, 2, 3], name=0).to_frame()
```

Comment:
Does the protocol allow duplicate names as well? I wonder what happens if by some depraved act a user has `columns=[0, "0"]` and goes through the protocol

Reply:
thanks for thinking about this - it currently gives a mysterious
```
    raise NotImplementedError(f"Columns of type {type(column)} not handled yet")
NotImplementedError: Columns of type <class 'pandas.core.frame.DataFrame'> not handled yet
```
good to give a good err msg here